### PR TITLE
Vcx add support for Postgres wallet

### PR DESCRIPTION
**Pull request by ianco**
Add support for postgres wallet.
Update python demo to make interactive and more functional.
Opening the PR for visibility to the work in progress.

### CODE REVIEW

**Comment by ianco** [comment]
Note also fixes the segfault that occurs after a successful send_proof()

File: vcx/wrappers/python3/demo/docker_pool_transactions_genesis [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{"reqSignature":{},"txn":{"data":{"data":{"alias":"Node1","blskey":"4N8aUNHSgjQVgkpm8nhNEfDf6txHznoYREg9kirmJrkivgL4oSEimFF6nsQ6M41QvhM2Z33nves5vfSn9n1UwNFJBYtWVnHYMATn76vLuL3zU88KyeAYcHfsih3He6UHcXDxcaecHVz6jhCYz1P2UZn2bDVruL5wXpehgBfBaLKm3Ba","blskey_pop":"RahHYiCvoNCtPTrVtP7nMC5eTYrsUA8WjXbdhNc8debh1agE9bGiJxWBXYNFbnJXoXhWFMvyqhqhRoq737YQemH5ik9oL7R4NTTCz2LEZhkgLJzB3QRQqJyBNyv7acbdHrAT8nQ9UkLbaVL9NBpnWXBTw4LEMePaSHEw66RzPNdAX1","client_ip":"10.0.0.2","client_port":9702,"node_ip":"10.0.0.2","node_port":9701,"services":["VALIDATOR"]},"dest":"Gw6pDLhcBcoQesN72qfotTgFa7cbuqZpkX3Xo6pLhPhv"},"metadata":{"from":"Th7MpTaRZVRYnPiabds81Y"},"type":"0"},"txnMetadata":{"seqNo":1,"txnId":"fea82e10e894419fe2bea7d96296a6d46f50f93f9eeda954ec461b2ed2950b62"},"ver":"1"}
```
[the remark this answers is not in the document]
Comment (ianco) [reply vcx/wrappers/python3/demo/docker_pool_transactions_genesis]: You're right this extra config is not necessary, I removed it.

**Comment by jovfer** [comment]
@glowkey could you review please?

File: vcx/dummy-cloud-agent/config.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (glowkey) [inline vcx/dummy-cloud-agent/config.json]: I think this file isn't needed, as it's just a copy of sample-config.json

File: vcx/libvcx/src/utils/libindy/wallet.rs [hunk 3]
```diff
@@ -14,15 +14,14 @@ pub static mut WALLET_HANDLE: i32 = 0;
 
 pub fn get_wallet_handle() -> i32 { unsafe { WALLET_HANDLE } }
 
-pub fn create_wallet(wallet_name: &str, wallet_type: Option<&str>) -> Result<(), u32> {
+pub fn create_wallet(wallet_name: &str, wallet_type: Option<&str>, storage_config: Option<&str>, storage_creds: Option<&str>) -> Result<(), u32> {
```
Comment (glowkey) [inline vcx/libvcx/src/utils/libindy/wallet.rs]: There's an instance of this in the tests for src/api/vcx.rs that hasn't been updated. (also other instances of delete_wallet and init_wallet).

Change Cargo.toml to have:

default = ["agency","pool_tests"]

To see the problems.

File: vcx/dummy-cloud-agent/config.json (hunk 2 above)
Comment (ianco) [reply vcx/dummy-cloud-agent/config.json]: File is removed.

File: vcx/libvcx/src/utils/libindy/wallet.rs (hunk 3 above)
Comment (ianco) [reply vcx/libvcx/src/utils/libindy/wallet.rs]: @glowkey   thanks for the review, files are updates.

**Pull request merged**